server/modules/protocol/MySQL/mysql_common.cc

```diff
@@ -1768,3 +1767,126 @@ void mxs_mysql_execute_kill_user(MXS_SESSION* issuer, const char* user, kill_typ
+{
+    char *trx_info, *var_name, *var_value;
+
+    if (GWBUF_LENGTH(buff) < 9  || !mxs_mysql_is_ok_packet(buff))
+    {
+        return;
+    }
+
+    if (!GWBUF_IS_CONTIGUOUS(buff))
+    {
+        buff = gwbuf_make_contiguous(buff);
+    }
+
+    buff->gwbuf_type |= GWBUF_TYPE_REPLY_OK;
+
+    uint8_t* ptr = GWBUF_DATA(buff);
+
+    ptr += (MYSQL_COM_OFFSET + 1); // Header and Command type
+
+    mxs_leint_consume(&ptr); // Affected rows
+    mxs_leint_consume(&ptr); // Last insert-id
+
+    uint16_t server_status = gw_mysql_get_byte2(ptr);
+    ptr += 2; // status      
+    ptr += 2; // number of warnings
+
+    if (server_capabilities & GW_MYSQL_CAPABILITIES_SESSION_TRACK)
+    {
+        if (ptr < buff->end) {
+            ptr += mxs_leint_consume(&ptr);  // info
+            if (server_status  & SERVER_SESSION_STATE_CHANGED)
+            {
+                mxs_leint_consume(&ptr);    // total SERVER_SESSION_STATE_CHANGED length
+                while (ptr < buff->end)
+                {
+                    enum_session_state_type type = (enum enum_session_state_type)mxs_leint_consume(&ptr);
+                    switch (type)
+                    {
+                        case SESSION_TRACK_SYSTEM_VARIABLES:
+                        case SESSION_TRACK_SCHEMA:
+                        case SESSION_TRACK_GTIDS:
+                        case SESSION_TRACK_TRANSACTION_CHARACTERISTICS:
+                            ptr += mxs_leint_consume(&ptr);
+                            break;
+                        case SESSION_TRACK_STATE_CHANGE:
+                            mxs_leint_consume(&ptr); //lenth 
+                            // system variables like autocommit, schema, charset ... 
+                            var_name = mxs_lestr_consume_dup(&ptr);
+                            var_value = mxs_lestr_consume_dup(&ptr);
+                            gwbuf_add_property(buff, var_name, var_value);
+                            MXS_FREE(var_name);
+                            MXS_FREE(var_value);
+                            break;
+                        case SESSION_TRACK_TRANSACTION_TYPE:
+                            mxs_leint_consume(&ptr); // length
+                            trx_info = mxs_lestr_consume_dup(&ptr);
+                            MXS_INFO("get trx_info:%s", trx_info);
+                            gwbuf_add_property(buff, (char *)"trx_state", trx_info);
+                            MXS_FREE(trx_info);
+                            break;
+                        default:
+                            ptr += mxs_leint_consume(&ptr);
+                            MXS_WARNING("recieved unexpecting session track type:%d", type);
+                            break;
+                    }
+                }
+            }
+        }
+    }
+}
+
+mysql_tx_state_t parse_trx_state(char *str)
+{
+    int s = TX_EMPTY;
+    assert(str);
+    do
+    {
+        switch (*str)
+        {
+            case 'T':
```

Comment:
Are these documented somewhere? If so, add a comment and a link to the documentation or document the meaning of these here.

Reply:
Tell Beyonce come get me